driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+/**
+ * Expresses a numeric value.
+ */
+public interface NumberExpression extends Expression {
+
+    NumberExpression multiply(NumberExpression n);
+
+    default NumberExpression multiply(final Number multiply) {
+        return this.multiply(Expressions.numberToExpression(multiply));
+    }
+
+    NumberExpression divide(NumberExpression n);
+
+    default NumberExpression divide(final Number divide) {
+        return this.divide(Expressions.numberToExpression(divide));
+    }
+
+    NumberExpression add(NumberExpression n);
+
+    default NumberExpression add(final Number add) {
+        return this.add(Expressions.numberToExpression(add));
+    }
+
+    NumberExpression subtract(NumberExpression n);
+
+    default NumberExpression subtract(final Number subtract) {
+        return this.subtract(Expressions.numberToExpression(subtract));
+    }
+
+    NumberExpression max(NumberExpression n);
+
+    NumberExpression min(NumberExpression n);
```

Comment:
Consider whether [reduce/min](https://github.com/mongodb/mongo-java-driver/pull/1053#discussion_r1037632332), getField(s), isArrayOr, and various other "non-secure" methods could all have a default value specified, or use Expression, for this initial release to have a "secured" API.

Reply:
Suggestion: we can use the term "unchecked" similarly to unchecked conversions<sup>1</sup> in Java, because this is exactly what we are doing here; there is no need to invent a new word (secure/safe) for the same thing.

---

<sup>1</sup> The correctness of unchecked conversions can't checked at runtime; i.e., they always succeed, but potentially lead to incorrect behavior later.

Reply:
These should be documented, and annotated, as of https://github.com/mongodb/mongo-java-driver/pull/1070